extensions/amp-carousel/0.1/slides.js

```diff
@@ -87,7 +87,14 @@ export class AmpSlides extends BaseCarousel {
   }
 
   /** @override */
-  goCallback(dir, animate) {
+  goCallback(dir, animate, opt_autoplay) {
+    // if `opt_autoplay` is not set, this means a user interacted with the
+    // the carousel and we cancel any current and future autoplay request
+    if (!opt_autoplay) {
```

Comment:
I think I'd rather see the button clicks as dedicated methods to that call cancel themselves.